fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java

```diff
@@ -139,9 +145,21 @@ public boolean hasPermission(final Context context, final Path absPath,
             return false;
         }
 
-        // this permission is required for login
         if (absPath == null) {
-            return actions.length == 1 && "read".equals(actions[0]);
+            // this permission is required for login
+            if (actions.length == 1 && READ.equals(actions[0])) {
+                return true;
+            } else {
+                // The REGISTER_NAMESPACE action and the REGISTER_TYPE action don't include
+                // a path and are allowed for all users.  The fedora 4 code base doesn't expose
+                // any endpoing that *JUST* registers a namespace or type, so the operations
```

Comment:
endpoint, although endpoing really does sound like a word. We should make up a meaning for it.